salt/utils/process.py

```diff
@@ -807,11 +807,15 @@ def _handle_signals(self, signum, sigframe):
         msg += '. Exiting'
         log.debug(msg)
         if HAS_PSUTIL:
-            process = psutil.Process(self.pid)
-            if hasattr(process, 'children'):
-                for child in process.children(recursive=True):
-                    if child.is_running():
-                        child.terminate()
+            for process in psutil.process_iter():
```

Comment:
Couldn't you do
```
if psutil.pid_exists(self.pid):
    process = psutil.Process(self.pid)
    if hasattr(process, 'children')
    try:
        ...
```
Instead of looping over every process on the system?

Reply:
Thanks for the tip, was waiting to see results of tests before cleaning up. Will be sure to implement it that way

Reply:
done